src/hotspot/share/gc/serial/serialHeap.hpp

```diff
@@ -116,29 +111,17 @@ class SerialHeap : public CollectedHeap {
                                bool   is_tlab,
                                bool   first_only);
 
-  // Helper function for two callbacks below.
-  // Considers collection of the first max_level+1 generations.
-  void do_collection(bool           full,
-                     bool           clear_all_soft_refs,
-                     size_t         size,
-                     bool           is_tlab,
-                     GenerationType max_generation);
-
-  // Callback from VM_GenCollectForAllocation operation.
-  // This function does everything necessary/possible to satisfy an
-  // allocation request that failed in the youngest generation that should
-  // have handled it (including collection, expansion, etc.)
-  HeapWord* satisfy_failed_allocation(size_t size, bool is_tlab);
-
-  // Callback from VM_GenCollectFull operation.
-  // Perform a full collection of the first max_level+1 generations.
   void do_full_collection(bool clear_all_soft_refs) override;
-  void do_full_collection(bool clear_all_soft_refs, GenerationType max_generation);
+  void do_full_collection_no_gc_locker(bool clear_all_soft_refs);
+
+  void collect_at_safepoint_no_gc_locker(bool full);
```

Comment:
I am not very convinced by the naming of the methods with the "no_gc_locker" constraint.  But I guess it is following same convention as "*at_safepoint" method naming.

Reply:
How about calling them `try_x` and `x` for the public and private API, respectively, e.g. `try_do_full_collection` and `do_full_collection`?

Reply:
I prefer that to the "no_gc_locker" emphasizing names

Reply:
I can do that for `try_collect_at_safepoint`, but `SerialHeap::do_full_collection` is an API from `CollectedHeap`.

Reply:
yeah, i only meant change those that are not "override" and have the "no_gclocker" postfix.